lib/cipherstash/index/bloom_filter.rb

```diff
@@ -0,0 +1,87 @@
+require_relative "../client/error"
+
+module CipherStash
+  class Index
+    # A bloom filter implementation designed to be used with the *FilterMatch index classes.
+    #
+    # @private
+    class BloomFilter
+      FILTER_TERM_BITS_MIN = 3
+      FILTER_TERM_BITS_MAX = 16
+      FILTER_TERM_BITS_DEFAULT = 3
+      FILTER_SIZE_DEFAULT = 256
+      VALID_FILTER_SIZES = [32, 64, 128, 256, 512, 1024, 2048, 4096, 8192, 16384, 32768, 65536]
+
+      # The "set" bits of the bloom filter
+      attr_reader :bits
+
+      # The size of the bloom filter in bits. Commonly referred to as "m". Since we only keep track of the set bits, the filter
+      # size determines the maximum value of the positions stored in the bits attr.
+      #
+      # Valid values are powers of 2 from 32 to 65536.
+      #
+      # @return [Integer]
+      attr_reader :filter_size
+
+      # The number of bits to set per term. Commonly referred to as "k".
+      #
+      # Valid values are integers from 3 to 16.
+      #
+      # @return [Integer]
+      attr_reader :filter_term_bits
```

Comment:
Ah fair, I mention this in [the tests](https://github.com/cipherstash/ruby-client/blob/5d85bf14d9b6ab71fc4afc9562db91e6b2c8210c/spec/cipherstash/index/bloom_filter_spec.rb#L62-L63) but the docs here don't reflect that. I'll reword the docs here.

Reply:
I think the documentation is *technically* correct (which is, of course, [the best kind of correct](https://www.youtube.com/watch?v=aIzMuPMicGc)), because "setting the same bit twice" is still "setting two bits".  Not that I'm against fixing the docs to make things clearer, I just enjoy being needlessly pedantic.